Build failure: Qtile

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *Qtile*

### Build log

This log comes from a system update (nixos-rebuild switch --upgrade).

```building
Executing setuptoolsBuildPhase
running tests
check flags: SHELL=/nix/store/k32c6vzr9g1nln6v0gypz6ar6lqjb63l-bash-5.2-p15/bin/bash TLP_NO_INIT=1 TLP_WITH_ELOGIND=0 TLP_WITH_SYSTEMD=1 DESTDIR=/nix/store/prq0558c0rw91j2xk5xf7hw9dinr1x5z-tlp-1.5.0 TLP_BATD=/share/tlp/bat.d TLP_BIN=/bin TLP_CONFDEF=/share/tlp/defaults.conf TLP_CONFREN=/share/tlp/rename.conf TLP_FLIB=/share/tlp/func.d TLP_MAN=/share/man TLP_META=/share/metainfo TLP_SBIN=/sbin TLP_SHCPL=/share/bash-completion/completions TLP_TLIB=/share/tlp VERBOSE=y checkall
*** checkbashisms ***************************************************************************
Failed to find pulseaudio headers. PulseVolume widget will be unavailable
/nix/store/8zx4h7r5mn1b913hgi8rwjfynwg1wgdi-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/config/setupcfg.py:520: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
  warnings.warn(msg, warning_class)
/nix/store/8zx4h7r5mn1b913hgi8rwjfynwg1wgdi-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/__init__.py:85: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
  dist.fetch_build_eggs(dist.setup_requires)
/nix/store/8zx4h7r5mn1b913hgi8rwjfynwg1wgdi-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/config/setupcfg.py:520: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
  warnings.warn(msg, warning_class)
running bdist_wheel
running build
running build_py
file libqtile/_ffi_pango.py (for module libqtile._ffi_pango) not found
file libqtile/backend/x11/_ffi_xcursors.py (for module libqtile.backend.x11._ffi_xcursors) not found
creating build
creating build/lib.linux-x86_64-cpython-310
creating build/lib.linux-x86_64-cpython-310/libqtile
copying libqtile/__init__.py -> build/lib.linux-x86_64-cpython-310/libqtile
creating build/lib.linux-x86_64-cpython-310/libqtile/backend
creating build/lib.linux-x86_64-cpython-310/libqtile/backend/x11

... (copying python code)

running egg_info
creating qtile.egg-info
writing qtile.egg-info/PKG-INFO
writing dependency_links to qtile.egg-info/dependency_links.txt
writing entry points to qtile.egg-info/entry_points.txt
writing requirements to qtile.egg-info/requires.txt
writing top-level names to qtile.egg-info/top_level.txt
writing manifest file 'qtile.egg-info/SOURCES.txt'
reading manifest file 'qtile.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files found matching '.coveragerc'
warning: no previously-included files found matching '.pylintrc'
warning: no previously-included files found matching 'tox.ini'
warning: no previously-included files found matching 'requirements.txt'
warning: no previously-included files found matching 'requirements-dev.txt'
warning: no previously-included files found matching 'libqtile/_ffi*.py'
warning: no previously-included files found matching 'libqtile/backend/x11/_ffi*.py'
warning: no previously-included files found matching 'Makefile'
warning: no previously-included files found matching 'dev.sh'
warning: no previously-included files found matching 'logo.png'
warning: no previously-included files found matching '.readthedocs.yaml'
no previously-included directories found matching 'bin'
no previously-included directories found matching 'docs'
no previously-included directories found matching 'scripts'
no previously-included directories found matching 'rpm'
warning: no previously-included files matching '__pycache__' found under directory '*'
warning: no previously-included files matching '*.py[co]' found under directory '*'
warning: no previously-included files matching '*' found under directory 'stubs'
adding license file 'LICENSE'
writing manifest file 'qtile.egg-info/SOURCES.txt'
copying libqtile/py.typed -> build/lib.linux-x86_64-cpython-310/libqtile
creating build/lib.linux-x86_64-cpython-310/libqtile/resources/battery-icons

... (copying icons)

file libqtile/_ffi_pango.py (for module libqtile._ffi_pango) not found
file libqtile/backend/x11/_ffi_xcursors.py (for module libqtile.backend.x11._ffi_xcursors) not found
generating cffi module 'build/lib.linux-x86_64-cpython-310/libqtile/_ffi_pango.py'
generating cffi module 'build/lib.linux-x86_64-cpython-310/libqtile/backend/x11/_ffi_xcursors.py'
running build_ext
generating cffi module 'build/temp.linux-x86_64-cpython-310/libqtile.backend.wayland._libinput.c'
creating build/temp.linux-x86_64-cpython-310
building 'libqtile.backend.wayland._libinput' extension
creating build/temp.linux-x86_64-cpython-310/build
creating build/temp.linux-x86_64-cpython-310/build/temp.linux-x86_64-cpython-310
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/nix/store/q88zzcq9dwfy4kq644p584a1z9vam6mv-libxcrypt-4.4.33/include -fPIC -DWLR_USE_UNSTABLE -I/usr/include/pixman-1 -I/nix/store/pikwdlha96c8a908plpcjylha230i2d7-python3.10-pywlroots-0.16.1/lib/python3.10/site-packages/wlroots/include -I/nix/store/0n4y44dnaxafqs7cg625aldrb152x7bx-python3-3.10.10/include/python3.10 -c build/temp.linux-x86_64-cpython-310/libqtile.backend.wayland._libinput.c -o build/temp.linux-x86_64-cpython-310/build/temp.linux-x86_64-cpython-310/libqtile.backend.wayland._libinput.o
build/temp.linux-x86_64-cpython-310/libqtile.backend.wayland._libinput.c:580:10: fatal error: wlr/types/wlr_damage_ring.h: No such file or directory
  580 | #include <wlr/types/wlr_damage_ring.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
error: command '/nix/store/rj91g41smpmp92xvhx6rjxz6878bn2b4-gcc-wrapper-12.2.0/bin/gcc' failed with exit code 1
/nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1593: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/j87hlprhdvh3j9vwqsw1qm3k01a93pzh-python3.10-qtile-0.22.1.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/bd2fh5171vr9m8512kay5xid2ql8239p-qtile-0.22.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zdpz0zasx1bwwcjcqn7vi4m55q9xs4m7-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cal0vcbghg7973rwk2ld94xvqmklr72q-nixos-system-thinkbook-16-plus-nixos-23.05pre464727.ac718d02867.drv' failed to build
```

### Additional context

I have been using Qtile for a long time, even before I recently changed to NixOS from Gentoo, and I am quite sure that Qtile currently needs <wlroots-0.16.0, i.e. wlroots-0.15.x. A few days ago, in commit b60a1fef4af412cb7c456a3199bcdc25f6783c3b, pywlroots is updated from 0.15.x to 0.16, pinning wlroots to 0.16, and therefore breaks the dependency of Qtile.

### Notify maintainers

@kamilchm @chvp 

I notify both maintainers of Qtile and pywlroots, because this may require a change to both packages, providing a pywlroots_0_15 and using that for qtile.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.5, NixOS, 23.05 (Stoat), 23.05pre462605.5a1dc8acd97`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/python-modules/pywlroots/default.nix
+++ b/pkgs/development/python-modules/pywlroots/default.nix
@@ -15,6 +15,7 @@
 , xkbcommon
 , xorg
 , pytestCheckHook
+, qtile
 }:
 
 buildPythonPackage rec {
@@ -40,6 +41,8 @@ buildPythonPackage rec {
   '';
 
   pythonImportsCheck = [ "wlroots" ];
+
+  passthru.tests = { inherit qtile; };
 
   meta = with lib; {
     homepage = "https://github.com/flacjacket/pywlroots";

--- a/pkgs/development/python-modules/pywlroots/default.nix
+++ b/pkgs/development/python-modules/pywlroots/default.nix
@@ -9,7 +9,7 @@
 , pixman
 , pythonOlder
 , udev
-, wlroots_0_16
+, wlroots
 , wayland
 , pywayland
 , xkbcommon
@@ -19,19 +19,19 @@
 
 buildPythonPackage rec {
   pname = "pywlroots";
-  version = "0.16.1";
+  version = "0.15.24";
   format = "setuptools";
 
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-GFJw88N0313PudM0o/jOvtoE3u8kSoV+lYgKlNya4wM=";
+    hash = "sha256-TvYhxiAbK+mpcEE9y79WH96dzeDnvI0xPaUxSYQqyHE=";
   };
 
   nativeBuildInputs = [ pkg-config ];
   propagatedNativeBuildInputs = [ cffi ];
-  buildInputs = [ libinput libxkbcommon pixman xorg.libxcb udev wayland wlroots_0_16 ];
+  buildInputs = [ libinput libxkbcommon pixman xorg.libxcb udev wayland wlroots ];
   propagatedBuildInputs = [ cffi pywayland xkbcommon ];
   nativeCheckInputs = [ pytestCheckHook ];
 

